mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp

```diff
@@ -2505,7 +2505,11 @@ LogicalResult PrivateClauseOp::verify() {
              << "Did not expect any values to be yielded.";
     }
 
-    if (yieldedTypes.size() == 1 && yieldedTypes.front() == symType)
+    // TODO How can we check that 2 types are compatible without leaking
```

Comment:
That might be doable by working with dialect interfaces, I think.

Reply:
Probably, but I couldn't find any interfaces declared by FIR types that can help here. @kiranchandramohan is there a way to do that right now?

To clarify, given 2 types, for example: `fir.ptr<i32>` and `fir.ref<i32>`, I want to test whether they are pointer/reference-like types that point/refer to the same underlying type.

Reply:
I'm talking about dialect interfaces, not type interfaces (similar to the `ConvertToLLVMPatternInterface`). These could be used to implement arbitrary dialect specific logic that this verifier could delegate to. Note that this might be a bit of overkill, though.

Reply:
Can you add suitable `fir.convert` operations to enforce the matching type in the generated code?

Reply:
Done! Thanks for the suggestion.